### PR TITLE
[ci] #1658: Add documentation check

### PR DESCRIPTION
**Pull request by ales-tsurko**
Sorry for omitting the PR template here, but the change is very little and clear. It just adds  documentation check (using `cargo doc`) to CI.

### CODE REVIEW

**Review by pesterev — CHANGES_REQUESTED** [review]
According to the CI report, a few comments need to be corrected to pass the CI checks.

File: ffi/src/lib.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ pub enum FfiResult {
     Ok = 0_i32,
 }
 
-/// Implement [`Handle`] for given types with first argument as the initial handle id.
```
Comment (outoftardis) [inline ffi/src/lib.rs:33]: @ales-tsurko this is the only link that is still unresolved.
I think my knowledge of Rust is limited here to understand why all the syntax I'm trying is failing, so if you have any ideas how to fix this -- please do

Comment (ales-tsurko) [reply ffi/src/lib.rs:33]: The problem caused by the `Handle` trait defined inside macro, so there's no trait definition until the macro is used. We have two options: 
1. define the trait outside the macro;
2. don't link documentation for the trait.

@appetrosyan what do you think?

Comment (outoftardis) [reply ffi/src/lib.rs:33]: maybe we can omit the link altogether? the description would still be clear even without the link

Comment (appetrosyan) [reply ffi/src/lib.rs:33]: > don't link documentation for the trait.

Is fine by me. We should avoid defining non-private traits inside macros, though.

**Review by outoftardis — APPROVED** [review]

**Pull request merged**